apps/arweave/src/ar_mining_worker.erl

```diff
@@ -333,24 +370,24 @@ process_sub_chunk(chunk1, Candidate, SubChunk, State) ->
 process_sub_chunk(chunk2, Candidate, SubChunk, State) ->
 	#mining_candidate{ session_key = SessionKey } = Candidate,
 	Candidate2 = Candidate#mining_candidate{ chunk2 = SubChunk },
-	case cycle_sub_chunk_cache(Candidate2, {chunk2, SubChunk}, State) of
-		{{chunk1, Chunk1, H1}, State2} ->
+	case cycle_sub_chunk_cache(Candidate2, SubChunk, #{chunk2 => true}, State) of
+		{{<<>>, #{chunk1 := true, h1 := H1}}, State2} ->
+			ar_mining_hash:compute_h2(self(), Candidate2#mining_candidate{ h1 = H1 }),
+			%% Decrement 1 for chunk2:
+			%% we're computing h2 for a peer so chunk1 was not previously read or cached
+			%% on this node
+			State2;
+		{{Chunk, #{chunk1 := true, h1 := H1}}, State2} ->
 			ar_mining_hash:compute_h2(
-				self(), Candidate2#mining_candidate{ chunk1 = Chunk1, h1 = H1 }),
+				self(), Candidate2#mining_candidate{ chunk1 = Chunk, h1 = H1 }),
```

Comment:
See comment [below](https://github.com/ArweaveTeam/arweave/pull/697/files#r1949811181)

The reason we decremented 2 here even though we had only removed a single chunk from the cache, is we knew that we had finished hashing both chunk1 and chunk2 so they would shortly be garbage collected and purged from memory (even though only chunk1 had ever been added to the cache)

Does that scenario translate to the new logic?